ricecooker/chefs.py

```diff
@@ -281,10 +283,57 @@ def save_channel_tree_as_json(self, channel):
         self.CHEF_RUN_DATA['tree_archives']['current'] = filename.replace(os.getcwd() + '/', '')
         self.save_chef_data()
 
+    def save_channel_metadata_as_csv(self, channel):
+        # create data folder in chefdata
+        # config.LOGGER.info('SAVE METADATA TO CSV')
+        DATA_DIR = os.path.join('chefdata', 'data')
+        if not os.path.isdir(DATA_DIR):
+            os.makedirs(DATA_DIR)
+        metadata_csv = csv.writer(open(os.path.join(DATA_DIR, 'content_metadata.csv'), 'w', newline='', encoding='utf-8'))
+        metadata_csv.writerow(config.CSV_HEADERS)
+
+        channel.save_channel_children_to_csv(metadata_csv)
+
+    def load_channel_metadata_from_csv(self, metadata_csv = None, metadata_dict = {}):
```

Comment:
`meatadata_dict = {}` will retain modifications between calls. You can use `dict()` instead, much like the `list()` fix. 

Also, are there cases where we want to pass these variables in to the function? If not, we can just make this function take zero args?

Reply:
You are right. Function no longer takes any arguments